x/launch/keeper/msg_server_create_chain.go

```diff
@@ -35,6 +36,18 @@ func (k msgServer) CreateChain(goCtx context.Context, msg *types.MsgCreateChain)
 		return nil, spnerrors.Criticalf("cannot create the chain: %v", err.Error())
 	}
 
+	if msg.HasCampaign {
+		campaign, found := k.campaignKeeper.GetCampaignChains(ctx, msg.CampaignID)
+		if !found {
+			campaign = campaigntypes.CampaignChains{
+				CampaignID: msg.CampaignID,
+				Chains:     []uint64{},
+			}
+		}
+		campaign.Chains = append(campaign.Chains, id)
+		k.campaignKeeper.SetCampaignChains(ctx, campaign)
+	}
+
```

Comment:
```suggestion
```

The last argument of `CreateNewChain` are `hasCampaign`, `campaignID` and `isMainnet`. We must provide those values to this function with mainnet = false. The chain is added to the list in `cmapaign` inside this function